### PR TITLE
[Merged by Bors] - chore(Data/Quot): rename `surjective_*_mk` => `*.mk_surjective`

### PR DESCRIPTION
**Pull request by astrainfinita**
`surjective_quot_mk` => `Quot.mk_surjective`
`surjective_quotient_mk` => `Quotient.mk_surjective`
`surjective_quotient_mk'` => `Quotient.mk'_surjective`
`Quotient.surjective_Quotient_mk''` => `Quotient.mk''_surjective`

This is consistent with the naming of most lemmas about `Function.Surjective` in mathlib.

Also make `Setoid` parameters implicit.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by urkud** [comment]
Many lemmas in Mathlib use `*_surjective` instead of `surjective_*`.

File: Mathlib/Data/Quot.lean [hunk 1]
```diff
@@ -319,15 +319,32 @@ theorem Quotient.liftOn₂_mk {α : Sort*} {β : Sort*} {_ : Setoid α} (f : α
...
+  Quot.exists_rep
+
+/-- `Quotient.mk'` is a surjective function. -/
+theorem Quotient.surjective_mk' [s : Setoid α] :
+    Function.Surjective (Quotient.mk' : α → Quotient s) :=
+  Quot.exists_rep
+
+/-- `Quot.mk r` is a surjective function. -/
+@[deprecated Quot.surjective_mk (since := "2024-09-02")]
 theorem surjective_quot_mk (r : α → α → Prop) : Function.Surjective (Quot.mk r) :=
```
Comment (eric-wieser) [inline Mathlib/Data/Quot.lean:337]: Arguably you should use `deprecated alias` for these, which also adjusts the docstring. Maybe it's not worth the risk of changing the statement by accident though.

**Review by eric-wieser — COMMENTED** [review]
It's interesting that we have https://leanprover-community.github.io/contribute/naming.html#injectivity but no corresponding rule for naming things `mk_surjective` instead of `surjective_mk`. It might be worth running a quick zulip poll to see if there's consensus.

**Review by eric-wieser — APPROVED** [review]
bors merge

Thanks!

**Pull request closed without merging**